src/test/java/com/ot/restaurant/repositories/CustomerRepositoryImplTest.java

```diff
@@ -0,0 +1,91 @@
+package com.ot.restaurant.repositories;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import com.ot.restaurant.entities.Customer;
+import com.ot.restaurant.exceptions.IdNullException;
+import com.ot.restaurant.fixtures.CustomerBuilder;
+import com.ot.restaurant.fixtures.CustomerFixture;
+import constants.Status;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class CustomerRepositoryImplTest {
+  private CustomerRepositoryImpl customerRepository = new CustomerRepositoryImpl();
+  private List<Customer> customerListTest = new ArrayList<>();
+
+  @BeforeEach
+  void setUp() throws Exception {
+
+    final var customers =
+        List.of(
+            CustomerFixture.buildCustomerFromExample(
+                new CustomerBuilder().withFirstName("Loki1").build()),
+            CustomerFixture.buildCustomerFromExample(
+                new CustomerBuilder().withFirstName("Loki2").build()),
+            CustomerFixture.buildCustomerFromExample(
+                new CustomerBuilder().withFirstName("Loki3").build()));
+    customers.forEach(customerRepository::insert);
+    customerListTest.addAll(customers);
+  }
+
+  @Test
+  void shouldReturnTheCustomerListWhenFindAll() {
+    assertEquals(customerListTest, customerRepository.findAll());
+  }
+
+  @Test
+  void shouldReturnSpecificCustomerWhenFindByIdAndStatusActive() throws Exception {
+    Customer customer = new Customer();
+    customer.setFirstName("Samahia");
+    customer.setStatus(Status.DELETED);
+    customerRepository.insert(customer);
+    final var existingCustomer = customerRepository.findById(customer.getId(), Status.ACTIVE).get();
```

Comment:
At this point you don't know the id of the customer, this only works now due to reference behavior, in this case you will need to iterate through all the customers and look for the one you just added.

Another suggestion is, why don't we find one of the existing customers in the repo, instead of adding a new one?

you could do something as simple as:

```
final var existingCustomer = customerRepository.findById(1L, Status.ACTIVE).get();

assertEquals(1L, existingCustomer.getId());
assertEquals("Loki1", existingCustomer.getName());
```